extension/src/openvic-extension/singletons/MapItemSingleton.cpp

```diff
@@ -0,0 +1,251 @@
+	}
+
+	return billboard;
+}
+
+// repackage the billboard object into a godot dictionnary for the Billboard manager to work with
+bool MapItemSingleton::add_billboard_dict(std::string_view name, TypedArray<Dictionary>& billboard_dict_array) const {
+
+	static const StringName name_key = "name";
+	static const StringName texture_key = "texture";
+	static const StringName scale_key = "scale";
+	static const StringName noOfFrames_key = "noFrames";
+
+	GFX::Billboard const* billboard = get_billboard(name,false);
+
+	ERR_FAIL_NULL_V_MSG(
+		billboard, false, "Failed to find billboard"
+	);
+
+	Dictionary dict;
+
+	dict[name_key] = Utilities::std_to_godot_string(billboard->get_name());
+	dict[texture_key] = Utilities::std_to_godot_string(billboard->get_texture_file());
+	dict[scale_key] = billboard->get_scale().to_float();
+	dict[noOfFrames_key] = billboard->get_no_of_frames();
+
+	billboard_dict_array.push_back(dict);
+
+	return true;
+}
+
+
+//get an array of all the billboard dictionnaries
+TypedArray<Dictionary> MapItemSingleton::get_billboards() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, {});
+
+	TypedArray<Dictionary> ret;
+
+	for (std::unique_ptr<GFX::Object> const& obj : game_singleton->get_definition_manager().get_ui_manager().get_objects()) {
+		if (obj->is_type<GFX::Billboard>()) {
+			add_billboard_dict(obj->get_name(), ret);
+		}
+	}
+	
+	return ret;
+}
+
+//dont use this, it includes water provinces
+int32_t MapItemSingleton::get_province_count() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, 0);
+
+	return game_singleton->get_definition_manager().get_map_definition().get_province_definition_count();
+}
+
+PackedVector2Array MapItemSingleton::get_province_positions() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedVector2Array());
+
+	PackedVector2Array billboard_pos {};
+	
+	for(ProvinceDefinition const& prov : game_singleton->get_definition_manager().get_map_definition().get_province_definitions()){
+		if(prov.is_water()) continue; //billboards dont appear over water, skip
+
+		fvec2_t city_pos = prov.get_city_position();
+		Vector2 pos = Utilities::to_godot_fvec2(city_pos) / game_singleton->get_map_dims();
+		billboard_pos.push_back(pos);
+
+	}
+	
+	return billboard_pos;
+}
+
+//includes non-existent countries, used for setting the billboard buffer size
+int32_t MapItemSingleton::get_capital_count() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, 0);
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, 0);
+
+	return instance_manager->get_country_instance_manager().get_country_instance_count();
```

Comment:
```suggestion
	return game_singleton->get_definition_manager().get_country_definition_manager().get_country_definition_count();
```
Use this instead, it gives the same number without needing to worry about instance manager being initialised